src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -203,8 +201,8 @@ public AzureVMDeploymentInfo createDeployment(
             final String storageAccountType = template.getStorageAccountType();
             final String diskType = template.getDiskType();
             final int osDiskSize = template.getOsDiskSize();
-            final String availabilityType = template.getAvailabilityType();
-            final String availabilitySet = template.getAvailabilitySet();
+            final AzureVMAgentTemplate.AvailabilityTypeClass availabilityType = template.getAvailabilityType();
+            final String availabilitySet = availabilityType.getAvailabilitySet();
```

Comment:
This may cause NullPointerException if the availability type has not been set.

Reply:
Fixed here and further down